llvm/test/CodeGen/AIE/aie2/internalize.ll

```diff
@@ -0,0 +1,123 @@
+; RUN: opt -O1 -S -mtriple=aie2-none-unknown-elf -aie2-internalize-symbols < %s | FileCheck --check-prefix=ONLYMAIN %s
+; RUN: opt -O1 -S -mtriple=aie2-none-unknown-elf -aie2-internalize-symbols -aie2-internalize-skip-functions="skipfunc1,skipfunc2,skipfunc3" < %s | FileCheck --check-prefix=FUNCLIST %s
+
+; ONLYMAIN-NOT: @gvar_unused
+; FUNCLIST-NOT: @gvar_unused
+@gvar_unused = addrspace(1) global i32 undef, align 4
+
+; ONLYMAIN: @gvar_used
+; FUNCLIST: @gvar_used
+@gvar_used = addrspace(1) global i32 undef, align 4
+
+; ONLYMAIN-NOT: define{{.*}}void @func_unused
+; FUNCLIST-NOT: define{{.*}}void @func_unused
+define void @func_unused(ptr addrspace(1) %out, i32 %val) {
+entry:
+  store volatile i32 %val, ptr addrspace(1) %out
+  ret void
+}
+
+; ONLYMAIN: define{{.*}}void @func_used_noinline
+; FUNCLIST-NOT: define{{.*}}void @func_used_noinline
+define void @func_used_noinline(ptr addrspace(1) %out, i32 %val) #0 {
+entry:
+  store volatile i32 %val, ptr addrspace(1) %out
+  ret void
+}
+
+; ONLYMAIN-NOT: define{{.*}}void @func_used_alwaysinline
+; FUNCLIST-NOT: define{{.*}}void @func_used_alwaysinline
+define void @func_used_alwaysinline(ptr addrspace(1) %out, i32 %val) #1 {
+entry:
+  store volatile i32 %val, ptr addrspace(1) %out
+  ret void
+}
+
+; ONLYMAIN: declare void @decl_func()
+; FUNCLIST-NOT: declare void @decl_func()
+declare void @decl_func()
+
+; ONLYMAIN: define{{.*}}void @main()
+; ONLYMAIN: tail call fastcc void @func_used_noinline
+; ONLYMAIN: store volatile
+; ONLYMAIN: tail call fastcc void @decl_func()
+; ONLYMAIN: ret void
+; FUNCLIST-NOT: define{{.*}}void @main()
+; FUNCLIST-NOT: tail call fastcc void @func_used_noinline
+; FUNCLIST-NOT: store volatile
+; FUNCLIST-NOT: tail call fastcc void @decl_func()
+; FUNCLIST-NOT: ret void
+define void @main() {
+entry:
+    %val = add i32 42, 0
+    tail call fastcc void @func_used_noinline(ptr addrspace(1) @gvar_used, i32 %val)
+    tail call fastcc void @func_used_alwaysinline(ptr addrspace(1) @gvar_used, i32 %val)
+    tail call fastcc void @decl_func()
+    ret void
+}
+
+; ONLYMAIN-NOT: define{{.*}}void @func_used_skipfunc1
+; FUNCLIST: define{{.*}}void @func_used_skipfunc1
+define void @func_used_skipfunc1(ptr addrspace(1) %out, i32 %val) #0 {
+entry:
+  store i32 %val, ptr addrspace(1) %out
+  ret void
+}
+
+; ONLYMAIN-NOT: define{{.*}}void @skipfunc1()
+; ONLYMAIN-NOT: tail call fastcc void @func_used_skipfunc1
```

Comment:
Maybe just `call{{.*}}void @func_used_skipfunc1`? I'm nervous about all those attributes that can change when we bump the LLVM version. (Same for other check lines)

Reply:
yeah i should probably change that.